src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,800 @@
+            }
+
+            SocksCommandType commandType = socksRequestFW.command().get();
+            decodeCommandType(commandType,
+                socksRequestFW.buffer(),
+                socksRequestFW.offset(),
+                socksRequestFW.limit());
+        }
+
+        private void onSocksConnect(
+            SocksRequestFW socksRequest)
+        {
+            if (socksRequest == null)
+            {
+                //TODO
+            }
+
+            final MessagePredicate filter = (t, b, o, l) ->
+            {
+                if (t > 0 && t <= SignalFW.TYPE_ID && o < l)
+                {
+                    return true;
+                }
+                return false;
+            };
+
+            final RouteFW route = router.resolve(routeId, this.authorization, filter, wrapRoute);
+            if(route != null)
+            {
+                final long newRouteId = route.correlationId();
+                final long newInitialId = supplyInitialId.applyAsLong(newRouteId);
+                final long newReplyId = supplyReplyId.applyAsLong(newInitialId);
+
+                final MessageConsumer newTarget = router.supplyReceiver(newInitialId);
+                final SocksServerStream socksServerStream = new SocksServerStream(this, newTarget,
+                    newRouteId, newInitialId, newReplyId);
+
+                SocksBeginExFW socksBeginEx = socksBeginExRW.wrap(extBuffer,0, extBuffer.capacity())
+                                                            .typeId(socksTypeId)
+                                                            .address(socksRequest.address().domainName())
+                                                            .port(socksRequest.port())
+                                                            .build();
+
+                socksServerStream.doApplicationBeginEx(newTarget, newInitialId, decodeTraceId,
+                socksBeginEx.buffer(), socksBeginEx.offset(), socksBeginEx.limit());
+                correlations.put(newReplyId, socksServerStream);
+            }
+        }
+
+        private void onNetworkReset(
+            ResetFW reset)
+        {
+            final long traceId = reset.trace();
+            doReset(traceId);
+        }
+
+        private void onNetworkSignal(
+            SignalFW signal)
+        {
+            final long traceId = signal.trace();
+            doSignal(traceId);
+        }
+
+        private void onHandshakeRequest(
+            SocksHandshakeRequestFW onHandshakeRequest
+        )
+        {
+            if (onHandshakeRequest.version() != 5)
+            {
+                //TODO
+            }
+            int nmethods = onHandshakeRequest.nmethods();
+            OctetsFW method = onHandshakeRequest.methods();
+            if (nmethods != method.limit() - method.offset())
+            {
+                //TODO refuse connection
+            }
+            else
+            {
+                onHandshakeReply(method);
+            }
```

Comment:
`private void onHandshakeRequest(
            SocksHandshakeRequestFW handshakeRequest)
        {
            OctetsFW ipRO = handshakeRequest.methods();
            byte[] method = new byte[ipRO.sizeof()];
            ipRO.buffer().getBytes(ipRO.offset(), method, 0, ipRO.sizeof());

            if (handshakeRequest.version() != 5)
            {
                doNetworkEnd(supplyTraceId.getAsLong());
            }
            else if (method[SocksAuthenticationMethod.NO_AUTHENTICATION_REQUIRED.value()] == -1)
            {
                doSocksHandshakeReply(SocksAuthenticationMethod.NO_ACCEPTABLE_METHODS);
                doNetworkEnd(supplyTraceId.getAsLong());
            }
            else
            {
                doSocksHandshakeReply(SocksAuthenticationMethod.NO_AUTHENTICATION_REQUIRED);
                decodeState = this::decodeCommandType;
            }
        }`
I didn't find the method `indexOfByte` the code above is what I did, based on the method  name. Is that ok?

Reply:
That's right, you'll need to implement `indexOfByte` as a `private static` method.
Make sure _not_ to allocate any `byte[]`, since you can easily iterate directly over the `OctetsFW` `buffer` from `offset` to `limit` instead.

Reply:
I use the buffer but the value is a short rather than byte. Should I use indexOfShort?  https://github.com/JinL0/nukleus-socks.java/blob/8ab796f39872e000ff82d1c447c9dafbc7ccd0cc/src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java#L736

Reply:
I use the buffer but the value is a short rather than byte. Should I use indexOfShort?  https://github.com/JinL0/nukleus-socks.java/blob/8ab796f39872e000ff82d1c447c9dafbc7ccd0cc/src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java#L736

Reply:
According to the spec each method consumes 1 byte so you need to search the methods, 1 byte at a time, until you find one with the right value. 

Calling buffer.getShort(...) is not correct because that would assume each method takes 2 bytes, not 1 byte.